adafruit_ds18x20.py

```diff
@@ -0,0 +1,114 @@
+# The MIT License (MIT)
+#
+# Copyright (c) 2017 Carter Nelson for Adafruit Industries
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+"""
+`adafruit_ds18x20`
+====================================================
+
+Driver for Dallas 1-Wire temperature sensor.
+
+* Author(s): Carter Nelson
+"""
+
+__version__ = "0.0.0-auto.0"
+__repo__ = "https://github.com/adafruit/Adafruit_CircuitPython_DS18x20.git"
+
+import time
+from adafruit_onewire.device import OneWireDevice
+from micropython import const
+
+_CONVERT = const(0x44)
+_RD_SCRATCH = const(0xBE)
+_WR_SCRATCH = const(0x4E)
+_CONVERSION_TIMEOUT = const(1)
+RESOLUTION = (9, 10, 11, 12)
+
+class DS18X20(object):
+    """Class which provides interface to DS18X20 temperature sensor."""
+
+    def __init__(self, bus, address):
+        if address.family_code == 0x10 or address.family_code == 0x28:
+            self._address = address
+            self._device = OneWireDevice(bus, address)
+            self._buf = bytearray(9)
+        else:
+            raise Exception('Incorrect family code in device address.')
```

Comment:
Its generally bad to raise Exception because its subclassed by a bunch of things. That makes it hard to catch specific exceptions and not others. 

I2CDevice raises a ValueError [here](https://github.com/adafruit/Adafruit_CircuitPython_BusDevice/blob/master/adafruit_bus_device/i2c_device.py#L58). The built in list with explanations is [here](https://docs.python.org/3.4/library/exceptions.html).

Reply:
Thanks, yeah, I was wondering about that and was just replicating some of what was already in the code. Should be a ValueError - but are you OK with the idea of raising an exception in general?